components/salesforce_rest_api/sources/common-instant.mjs

```diff
@@ -0,0 +1,145 @@
+      this.setNameField(nameField);
+
+      // Create the webhook in the Salesforce platform
+      const secretToken = uuidv4();
+      let webhookData;
+      try {
+        webhookData = await this.salesforce.createWebhook(
+          this.http.endpoint,
+          this.objectType,
+          this.getEventType(),
+          secretToken,
+          {
+            fieldsToCheck: this.getFieldsToCheck(),
+            fieldsToCheckMode: this.getFieldsToCheckMode(),
+          },
+        );
+      } catch (err) {
+        console.log("Create webhook error:", err.response?.data ?? err);
+        throw err;
+      }
+      this._setSecretToken(secretToken);
+      this._setWebhookData(webhookData);
+    },
+    async deactivate() {
+      // Create the webhook from the Salesforce platform
+      const webhookData = this._getWebhookData();
+      await this.salesforce.deleteWebhook(webhookData);
+    },
+  },
+  methods: {
+    _getSecretToken() {
+      return this.db.get("secretToken");
+    },
+    _setSecretToken(secretToken) {
+      this.db.set("secretToken", secretToken);
+    },
+    _getWebhookData() {
+      return this.db.get("webhookData");
+    },
+    _setWebhookData(webhookData) {
+      this.db.set("webhookData", webhookData);
+    },
+    getNameField() {
+      return this.db.get("nameField");
+    },
+    setNameField(nameField) {
+      this.db.set("nameField", nameField);
+    },
+    _isValidSource(event) {
+      const webhookToken = event.headers["x-webhook-token"];
+      const secretToken = this._getSecretToken();
+      return webhookToken === secretToken;
+    },
+    processEvent(event) {
+      const { body } = event;
+      const meta = this.generateMeta(event);
+      this.$emit(body, meta);
+    },
+    generateMeta() {
+      throw new Error("generateMeta is not implemented");
+    },
+    getEventType() {
+      throw new Error("getEventType is not implemented");
+    },
+    /**
+     * This method returns the fields in the SObject type (e.g. Account, Lead, etc.) that the event
+     * source should listen for updates to. This base implementation returns `undefined`, to not
+     * necessitate any specific fields to be updated.
+     *
+     * @returns the fields in the SObject type for which to receive updates
+     */
+    getFieldsToCheck() {
+      return undefined;
+    },
+    /**
+     * This method returns whether the event source should listen for updates where `all` the fields
+     * in the SObject are updated, or when `any` of them are. This base implementation returns
+     * `undefined` to use to client's default `fieldToCheckMode` (`any`).
+     *
+     * @returns whether the webhook should receive events when `all` the fields to check are
+     * updated, or when `any` of them are
+     */
+    getFieldsToCheckMode() {
+      return undefined;
+    },
+  },
+  async run(event) {
+    if (!this._isValidSource(event)) {
+      this.http.respond({
+        statusCode: 404,
+      });
```

Comment:
I think we often respond with HTTP 404 to requests from unrecognized sources to notify the requester that its destination is incorrect. Some sources with this pattern: [ringcentral http-based](https://github.com/PipedreamHQ/pipedream/blob/master/components/ringcentral/sources/common/http-based.js#L95), [sentry issue-event](https://github.com/PipedreamHQ/pipedream/blob/54d9f846e62120a4f5539af09fc10d724bc086c6/components/sentry/sources/issue-event/issue-event.js#L59-L60), [gitlab new-branch](https://github.com/PipedreamHQ/pipedream/blob/54d9f846e62120a4f5539af09fc10d724bc086c6/components/gitlab/sources/new-branch/new-branch.js#L70-L73)

I could add a console.log message about skipping the event instead of throwing an error?
Ex:
https://github.com/PipedreamHQ/pipedream/blob/54d9f846e62120a4f5539af09fc10d724bc086c6/components/datadog/sources/new-monitor-event/new-monitor-event.js#L79-L84

Reply:
Added a `console.log`: https://github.com/PipedreamHQ/pipedream/pull/1984/commits/75306cfc1b75bf26eabb30504051a509e7ab02d3#diff-f43e31084ce9774f09cbc1b83017d745fb8e13a2ff71c46e29e1a82a25c7b891R121-R125